From 90cf16d88f74c73ff8760ff2f4c8067fe86700e8 Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@nxp.com>
Date: Tue, 8 May 2018 10:48:50 -0500
Subject: [PATCH 3739/5242] MLK-18237 XRDC: rpmsg: add power domain to avoid
 crash

commit  417653db4446ef0f6f6221cc898cd6ff2e18f687 from
https://source.codeaurora.org/external/imx/linux-imx.git

[    0.737561] Unhandled fault: synchronous external abort (0x96000210) at 0xffff00000c160000
[    0.745503] Internal error: : 96000210 [#1] PREEMPT SMP
[    0.750695] Modules linked in:
[    0.753739] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.9.88-04903-ga209cd8 #464
[    0.761118] Hardware name: Freescale i.MX8QXP MEK (DT)
[    0.766246] task: ffff80083a088000 task.stack: ffff80083a034000
[    0.772160] PC is at MU_Init+0x0/0x38
[    0.775805] LR is at imx_rpmsg_probe+0x22c/0x510

Signed-off-by: Frank Li <Frank.Li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 0bbd3fc..d47669a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -2801,6 +2801,7 @@
 		rpmsg: rpmsg{
 			compatible = "fsl,imx8qxp-rpmsg";
 			status = "disabled";
+			power-domains = <&pd_cm40_mu0a0>;
 		};
 	};
 
-- 
1.7.9.5

